src/rospack_backcompat.cpp

```diff
@@ -40,7 +40,8 @@ namespace rospack
 int
 ROSPack::run(int argc, char** argv)
 {
-  rospack::Rospack rp;
+  // allow caching of results between calls (of same process)
+  static rospack::Rospack rp;
```

Comment:
@dirk-thomas 
Thanks for discussing the issue. Please keep in mind during your discussions, that this PR aims for a caching behavior on existing code, primarily calls to [`ros::package::getPath()`](https://github.com/ros/ros/blob/jade-devel/core/roslib/src/package.cpp#L89). These calls go through [`roslib`](https://github.com/ros/ros/blob/jade-devel/core/roslib/src/package.cpp#L51) and thus don't provide the possibility to pass in a `ROSPack` instance.

What do you think of [my previous comment](https://github.com/ros/rospack/pull/49#discussion_r45936547)? Investigating the `getenv` calls in `rospack`, one can see, that the environment is evaluated for retrieving the `ROS_PACKAGE_PATH`, the cache path, and the cache timeout. As rospack re-crawls if `ROS_PACKAGE_PATH` has changed in the environment, I don't see the need to pass in a new rospack instance at all.

```
rospack.cpp:259:  char* rpp = getenv("ROS_PACKAGE_PATH");
rospack.cpp:1882:  char* ros_home = getenv("ROS_HOME");
rospack.cpp:1891:    char* home_drive = getenv("HOMEDRIVE");
rospack.cpp:1892:    char* home_path = getenv("HOMEPATH");
rospack.cpp:1902:      home_path = getenv("HOME");
rospack.cpp:1929:  char* rpp = getenv("ROS_PACKAGE_PATH");
rospack.cpp:2036:        char *rpp = getenv("ROS_PACKAGE_PATH");
rospack.cpp:2061:  const char *user_cache_time_str = getenv("ROS_CACHE_TIMEOUT");
rospack.cpp:2083:  const char* ros_package_path = getenv("ROS_PACKAGE_PATH");
```